### PR TITLE
Handle member expressions in jsx-props-no-multi-spaces

### LINKED ISSUE #1881: Cannot read property 'name' of undefined at getPropName
env: 
"eslint": "^5.0.0",
"eslint-plugin-react": "^7.10.0",

I got a error when use eslint --ext .js src; 
the error is happened in `eslint-plugin-react\lib\rules\jsx-props-no-multi-spaces.js getPropName `  function, this function like can't handle the Node type, and I console the Node type, there is

<code>
{
  type: 'JSXMemberExpression',
  start: 1243,
  end: 1252,
  loc:
   SourceLocation {
     start: Position { line: 43, column: 9 },
     end: Position { line: 43, column: 18 } },
  range: [ 1243, 1252 ],
  object:
   Node {
     type: 'JSXIdentifier',
     start: 1243,
     end: 1247,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 1243, 1247 ],
     name: 'Menu',
     _babelType: 'JSXIdentifier',
     parent: [Circular] },
  property:
   Node {
     type: 'JSXIdentifier',
     start: 1248,
     end: 1252,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 1248, 1252 ],
     name: 'Item',
     _babelType: 'JSXIdentifier',
     parent: [Circular] },
  _babelType: 'JSXMemberExpression',
  parent:
   Node {
     type: 'JSXOpeningElement',
     start: 1242,
     end: 1269,
     loc: SourceLocation { start: [Object], end: [Object] },
     range: [ 1242, 1269 ],
     attributes: [ [Object] ],
     name: [Circular],
     selfClosing: false,
     _babelType: 'JSXOpeningElement',
     parent:
      Node {
        type: 'JSXElement',
        start: 1242,
        end: 1325,
        loc: [Object],
        range: [Array],
        openingElement: [Circular],
        closingElement: [Object],
        children: [Array],
        _babelType: 'JSXElement',
        parent: [Object] } } }
</code>

### PR DESCRIPTION
**Pull request by alexzherdev**
Resolves #1881

### CODE REVIEW

File: lib/rules/jsx-props-no-multi-spaces.js [hunk 1]
```diff
@@ -27,12 +27,16 @@ module.exports = {
...
-        return sourceCode.getText(propNode.argument);
-      } else if (propNode.type === 'JSXIdentifier') {
-        return propNode.name;
+      switch (propNode.type) {
+        case 'JSXSpreadAttribute':
+          return sourceCode.getText(propNode.argument);
+        case 'JSXIdentifier':
+          return propNode.name;
+        case 'JSXMemberExpression':
+          return `${propNode.object.name}.${propNode.property.name}`;
```
Comment (alexzherdev) [inline lib/rules/jsx-props-no-multi-spaces.js]: This will not handle `Foo.Bar.Baz`, do we need that?

Comment (ljharb) [reply lib/rules/jsx-props-no-multi-spaces.js]: I think it should handle dots, yes - and if bracket notation is supported, that too (but i don’t think it is)

Comment (alexzherdev) [reply lib/rules/jsx-props-no-multi-spaces.js]: Do you mean `Foo['Bar']`? Yeah, looks like it's not supported according to Babel REPL.

Comment (ljharb) [reply lib/rules/jsx-props-no-multi-spaces.js]: To clarify; it should handle any number of nested member expressions.

Comment (alexzherdev) [reply lib/rules/jsx-props-no-multi-spaces.js]: I hope recursion is good enough for practical uses 😄

File: tests/lib/rules/jsx-props-no-multi-spaces.js [hunk 2]
```diff
@@ -114,5 +116,11 @@ ruleTester.run('jsx-props-no-multi-spaces', rule, {
     errors: [
       {message: 'Expected only one space between "Foo.Bar" and "baz"'}
     ]
+  }, {
+    code: '<Foobar.Foo.Bar.Baz.Qux.Quux.Quuz.Corge.Grault.Garply.Waldo.Fred.Plugh  xyzzy="thud" />',
```
Comment (ljharb) [inline tests/lib/rules/jsx-props-no-multi-spaces.js]: This test has been eaten by a grue.

**Pull request merged**